### PR TITLE
Take serializeKeys into account in containsKey at ClientMapProxy

### PR DESCRIPTION
**Pull request by sancar**
fixes https://github.com/hazelcast/hazelcast/issues/16472

Note that after adding the check to `checkKeyFormat` ,
ClientMapNearCacheBasicSlowTest starts to fail. It has tests
for containsKey. With the fix, it is passing now. So no new test is
needed.

related to https://github.com/hazelcast/hazelcast/issues/16458

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/DefaultNearCache.java [hunk 1]
```diff
@@ -222,6 +223,8 @@ public V tryPublishReserved(K key, V value, long reservationId, boolean deserial
     private void checkKeyFormat(K key) {
         if (!serializeKeys) {
             checkNotInstanceOf(Data.class, key, "key cannot be of type Data!");
+        } else {
+            checkInstanceOf(Data.class, key, "key must be of type Data!");
         }
```
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/DefaultNearCache.java]: This looks clearer:
```
if (serializeKeys) {
	checkInstanceOf(Data.class, key, "key must be of type Data!");
} else {
    checkNotInstanceOf(Data.class, key, "key cannot be of type Data!");        
}
```

**Comment by ahmetmircik** [comment]
EE near cache has its own implementation, not to break EE code, please run EE near cache tests before merging.

**Comment by sancar** [comment]
@ahmetmircik Thanks for the tip. I  have run all tests under `com.hazelcast.client.map.impl.nearcache` and  found a failing test for the Enterprise. Added one more commit for that into this pr.

File: hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientReplicatedMapProxy.java [hunk 2]
```diff
@@ -112,7 +112,11 @@ protected void onInitialize() {
     private void initNearCache() {
         NearCacheConfig nearCacheConfig = getContext().getClientConfig().getNearCacheConfig(name);
         if (nearCacheConfig != null) {
-            nearCache = getContext().getNearCacheManager(getServiceName()).getOrCreateNearCache(name, nearCacheConfig);
+            NearCacheConfig copyNearCacheConfig = new NearCacheConfig(nearCacheConfig);
+            //We don't have serializeKeys support on replicated map nearcache
+            copyNearCacheConfig.setSerializeKeys(false);
```
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientReplicatedMapProxy.java]: can we fail fast here instead of forcing it to be `false`?

**Pull request merged**